python/az/aro/azext_aro/_rbac.py

```diff
@@ -99,6 +134,44 @@ def assign_contributor_to_routetable(cli_ctx, master_subnet, worker_subnet, obje
         ))
 
 
+def has_network_contributor(cli_ctx, resource, object_id):
+    # If a list/set, recursively call this function with each subresource
+    if isinstance(resource, (list, set)):
```

Comment:
There is no reason to special case out a list vs. a single item. I suggest you just write the function assuming a list, and wrap any single item arguments in a list as needed.